crates/services/producer/src/block_producer/tests.rs

```diff
@@ -534,6 +540,41 @@ mod produce_and_execute_block_txpool {
     }
 }
 
+fn block_with_height(height: impl Into<BlockHeight>) -> Block {
+    let header = PartialBlockHeader {
+        consensus: ConsensusHeader {
+            height: height.into(),
+            ..Default::default()
+        },
+        application: ApplicationHeader {
+            da_height: DaBlockHeight::default(),
+            ..Default::default()
+        },
+    };
+    let mint = Transaction::Mint(Mint::default());
+    let txs = vec![mint];
+    let outbox_message_ids = vec![];
+    let event_inbox_root = Bytes32::default();
+    Block::new(header, txs, &outbox_message_ids, event_inbox_root).unwrap()
+}
+
+#[tokio::test]
+async fn produce_and_execute_predefined_block__happy() {
+    // given
+    let height = 1u32;
+    let block = block_with_height(height);
+    let executor = MockExecutorWithCapture::default();
+    let ctx = TestContext::default_from_executor(executor.clone());
+
+    let producer = ctx.producer();
+
+    // when
+    let result = producer.produce_and_execute_predefined(&block).await;
+
+    // then
+    assert!(result.is_ok());
```

Comment:
Verify the time, DA block height, gas price, and coinbase recipient are correct. We can do that via the capture.